ml_pipeline_engine/dag/graph.py

```diff
@@ -1,13 +1,39 @@
-import networkx as nx
 import itertools
+import typing as t
+
+import networkx as nx
+
+from ml_pipeline_engine.types import NodeId
 
 
 class DiGraph(nx.DiGraph):
 
-    def __init__(self, is_recurrent: bool = False, **kwargs):
+    def __init__(self, is_recurrent: bool = False, is_oneof: bool = False, **kwargs):
         super().__init__(**kwargs)
 
         self.is_recurrent = is_recurrent
+        self.is_oneof = is_oneof
 
     def __hash__(self):
         return hash(tuple(sorted(itertools.chain(*self.nodes.keys(), *self.edges.keys()))))
+
+
+def get_connected_subgraph(
+    dag: DiGraph,
+    source: NodeId,
+    dest: NodeId,
+    is_recurrent: bool = False,
+    is_oneof: bool = False,
+) -> DiGraph:
+    """
+    Получить связный подграф между двумя заданными узлами графа
+    """
+
+    if len(dag) == 1:
```

Comment:
If it == 1, should it even go into this method?

Reply:
The function is common for all methods related to graph changes. Thus, I'd leave it how it is